Content.Client/Singularity/Systems/SingularitySystem.cs

```diff
@@ -45,7 +45,7 @@ protected override void OnSingularityStartup(EntityUid uid, SingularityComponent
         base.OnSingularityStartup(uid, comp, args);
         if (TryComp<SpriteComponent>(uid, out var sprite))
         {
-            sprite.LayerMapReserveBlank(comp.Layer);
+            sprite.LayerMapReserveBlank(comp.SpriteLayerKey);
         }
     }
```

Comment:
You should be able to just remove `OnSingularityStartup` and `OnAppearanceChange` altogether

Reply:
fucking magic